attributes-kt/src/commonMain/kotlin/space/kscience/attributes/AttributesBuilder.kt

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2018-2023 KMath contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package space.kscience.attributes
+
+/**
+ * A safe builder for [Attributes]
+ *
+ * @param O type marker of an owner object, for which these attributes are made
+ */
+public class AttributesBuilder<out O> internal constructor(
+    private val map: MutableMap<Attribute<*>, Any?>,
+) : Attributes {
```

Comment:
If we prefer safety via immutability, the `AttributesBuilder` class should not implement `Attributes`. Otherwise, we can change the "immutable" `Attributes` instance by accidentally changing it as a `AttributesBuilder` instance.
```suggestion
) {
```